### ISSUE TITLE
🌍 no such file or directory for custom localePath

### ISSUE THREAD

**Issue by FrancisLeigh** [isaachinman/next-i18next#1116]
## Describe the bug
I'm setting up my Nextjs project with internationalization using this module.
I'm following the setup guidelines and I am having to customize my `localePath` as my translation files are placed into a different folder to what `next-i18n` expects.

## Occurs in next-i18next version
`"next-i18next": "^8.1.3"`

## Steps to reproduce
I'll show my code below.

## Expected behaviour
The correct translation files should be found in the directory described in `localePath`

## Screenshots
Folder structure
<img width="964" alt="Screenshot 2021-03-31 at 16 49 19" src="https://user-images.githubusercontent.com/10357801/113173178-0c310200-9241-11eb-8b0b-6df047ae4eb5.png">

## OS (please complete the following information)
 - Device: Macbook pro (16-inch, 2019), Catalina
 - Browser: Any

### Error
```bash
Error: ENOENT: no such file or directory, scandir '/Users/francisleigh/Documents/C/visage/src/localization/en'
```
### next.config.js
```javascript
const withTM = require('next-transpile-modules')(['maisonsport-common-ui', 'styled-components/native']);
const withImages = require('next-images');

const { i18n } = require('./next-i18next.config')

module.exports = withTM(
  withImages({
    webpack: (config) => {
      config.resolve.alias = {
        ...(config.resolve.alias || {}),
        // Transform all direct `react-native` imports to `react-native-web`
        'react-native$': 'react-native-web',
      }
      config.resolve.extensions = [
        '.web.js',
        '.web.ts',
        '.web.tsx',
        ...config.resolve.extensions,
      ]
      return config
    },
    i18n,
  })
);

```
### next-i18next.config.js
```javascript
const path = require('path');

module.exports = {
  i18n: {
    // These are all the locales you want to support in
    // your application
    locales: ['en', 'fr', 'it', 'de'],
    // This is the default locale you want to be used when visiting
    // a non-locale prefixed path e.g. `/hello`
    defaultLocale: 'en',
    // This is a list of locale domains and the default locale they
    // should handle (these are only required when setting up domain routing)
    // Note: subdomains must be included in the domain value to be matched e.g. "fr.example.com".
  },
  localePath: path.resolve('src/localization'),
  localeStructure: '{{lng}}',
}
```

Quick note: If I set `localeExtension: ''` and set `localeStructure: '{{lng}}.json` it ends up looking for `/en.json.` and If i set `localeExtension: 'json'` - get the same error.

**Comment by isaachinman** [comment]
The name of the file is the namespace. I think you should be able to use `localeStructure: '{{ns}}.json`. Let me know if that works.

**Comment by FrancisLeigh** [comment]
Wow thanks for the swift reply 

That has not worked

###
```bash
Default namespace not found at /Users/francisleigh/Documents/C/visage/src/localization/common.json.json
```

**Comment by isaachinman** [comment]
@FrancisLeigh You'll need to play with your config values, but it should be possible to achieve what you want. I think just `{{ns}}`?

**Comment by FrancisLeigh** [comment]
`{{ns}}` is the `common` whereas `{{lng}}` is the language 

As you can see in my screenshot I need the long but its the localeExtension keeps putting a `.` on the end
I can't seem to achieve what I need.

**Comment by isaachinman** [comment]
Where are your translations coming from? Is there a reason you can't change the dir structure?

**Comment by FrancisLeigh** [comment]
we have a shared module that gets the translations from a google sheet and places them in that directory so to change the destination would require further code changes on multiple apps/

**Comment by isaachinman** [comment]
I recommend having a look at [these docs](https://www.i18next.com/principles/translation-resolution). Ultimately `next-i18next` just passes config options into `i18next`, so you should be able to achieve what you need.

**Comment by FrancisLeigh** [comment]
I think the issue is that regardless of wether or not you set `localeExtension` to `''`
it still appends a `.` to the end of the `localeStructure` but maybe we could change the config a bit so if we set `localeExtension` to `null` maybe it is negated completely. 

that way I could do 
```
localePath: path.resolve('src/localization'),
localeStructure: '{{lng}}.json',
localExtension: null,
```

**Comment by isaachinman** [comment]
Sure, want to work on a PR? Just keep in mind we'll need test coverage.

**Comment by mcha-dev** [comment]
are namespaces mandatory i.e. having `locales/en/common.json`? just to understand/confirm, currently we cannot just have `locales/en.json`?

**Comment by FrancisLeigh** [comment]
@mcha-dev yes from what i have seen you have to go by the common.json route. 

That said it *should* be customizable however the code just doesn't seem to work.

**Comment by isaachinman** [comment]
Whatever is possible within the i18next paradigm should be possible with `next-i18next`. If you find functionality which is not supported, PRs are welcomed.

**Comment by FrancisLeigh** [comment]
For the time being i have altered by node-script to be customizable i.e I am now able to get my language files in a `{{lng}}/common.json` format.

I'll take a look raising a PR later down the line when I get min

**Comment by H3RSKO** [comment]
The issue is that it is looking for a folder where you are trying to provide a file. If you edit your local next-i18next package commonjs > config > createConfig line 177, and remove the `.concat(locale)` at the end of the line it will work. 

I put in a PR with the fix.

**Comment by isaachinman** [comment]
@H3RSKO Your PR (#1236) is currently broken.

**Comment by H3RSKO** [comment]
Yeah, I assume it has to do with the default vs custom file layouts.

@FrancisLeigh and I have our layout set to `localPath/file.json`, whereas the library is seems to be looking for `localPath/lang/file.json`. I can't switch my file layout since I am hooking into an existing translation structure using [weblate](https://weblate.org/), but getting rid of that piece of code works for my use case. I guess it is breaking the app by the standard use case. 

Unfortunately I don't have the time to work on a fix right now, but I may try to revisit later.

**Comment by isaachinman** [comment]
Fixed by #1593
